stix_shifter_modules/vectra/stix_transmission/connector.py

```diff
@@ -0,0 +1,212 @@
+from .api_client import APIClient
+import json
+
+
+class Connector(BaseJsonSyncConnector):
+    VECTRA_MAX_PAGE_SIZE = 5000
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+
+    async def ping_connection(self):
+        return_obj = {}
+        response_dict = {}
+        try:
+            response_wrapper = await self.api_client.ping_data_source()
+            response_code = response_wrapper.code
+            if response_code == 200:
+                return_obj['success'] = True
+            elif response_code in (401, 403):
+                response_dict['code'] = response_code
+                response_dict['message'] = "Invalid Authentication"
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except Exception as ex:
+            if "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error when getting search results: %s', str(ex))
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    async def create_results_connection(self, query, offset, length):
+        """
+         Function to create query connection
+        :param query: str
+        :param offset: int
+        :param length: str
+        :return: list
+        """
+        response_dict = {}
+        return_obj = {}
+        data = []
+        try:
+            offset = int(offset)
+            length = int(length)
+            pagination_offset = offset
+            total_records = length
+            page = 1
+
+            # changing offset to page number
+            if offset >= length:
+                pagination_offset = offset % length
+                page = (offset // length) + 1
+
+            # set total record count
+            if self.api_client.result_limit < total_records:
+                total_records = self.api_client.result_limit
+
+            # set page size
+            if total_records <= Connector.VECTRA_MAX_PAGE_SIZE:
+                page_size = total_records
+            else:
+                page_size = Connector.VECTRA_MAX_PAGE_SIZE
+
+            query = f"page_size={page_size}&{query}&page={page}"
+            response_wrapper = await self.api_client.get_search_results(query)
+            response_code = response_wrapper.code
+            response = json.loads(response_wrapper.read().decode('utf-8'))
+
+            if response_code == 200:
+                data += response['results']
+                next_url = response.get('next')
+                while next_url:
+                    if len(data) >= total_records:
+                        break
+                    next_url = next_url.split('/api/v2.4/search/detections/?')
```

Comment:
why do you need to split() the next_url? you can just directly make api call using the next_url instead of this extra steps.

Reply:
next_url contains host, path, and query with pagination parameters.
self.api_client.get_search_results function expects only query, so we are splitting the next_url and passing the query with pagination parameters.
Host and path details were already configured in the api_client.py file, it requires only a query input.
Reuse the function, we are splitting the next_url.

Reply:
its just redundant to split up the url and reconstruct it for the api call. you can just make api call with the next url here without all this extra steps- `self.client.call_api()`

Reply:
Example for the next URL:
'https://host.com/api/v2.4/search/detections/?page_size=100&query_string=query&page=2'

next url contains
host name= host.com 
endpoint= /api/v2.4/search/detections/ 
pagination_parameters= page_size=100&query_string=query&page=2


If we directly use the "self.client.call_api()" function, even then we should split the host and pass the remaining parameters.
Making an api call using self.client.call_api(), we should only pass the endpoint and parameters, not the host name.
While initializing self.client, we are setting the host name.
self.client = RestApiClientAsync(connection.get('host'), port=None, headers=self.headers)

It is similar to the way we are handling in next_url currently.

Reply:
you can keep the current implementation to handle pagination api call.